platform/engineering/request-for-change/2019-09-30-cms-integration.md

```diff
@@ -0,0 +1,230 @@
+
+**Example**
+1. Somebody merges a branch into `master`
+1. A full Jenkins build is triggered, which then passes
+1. A broken link is introduced in the CMS
+1. Somebody else merges a branch into `master`
+1. A new full Jenkins build is triggered
+1. The Metalsmith script queries for the latest content with the
+   broken link
+1. Because of this latest content, the `vets-website` `master` build
+   fails, but not because of any changes in `vets-website` code
+
+The goal of this RFC is to determine an approach for separating the
+content build from the `vets-website` Webpack build, so when
+`vets-website` pulls the content to serve up, it's already been built
+and validated. **Build failures due to content issues will be isolated
+to the content build and will not affect `vets-website` builds or
+deploys.** This will improve the overall stability of the
+`vets-website` CI.
+
+## Design
+_Explain the proposed changes in enough detail so that a team member
+with working knowledge of the project could implement the change
+themselves. Bonus points if this is easily mappable to user stories or
+specs._
+
+I propose rearchitecting the `vets-website` build to separate the
+static content building into a process distinct from the website.
+
+![separate build diagram](images/separated-content-build-proposal.png)
+
+**Note:** Pulling the already-built content _can still_ have two
+different results, much like the current state, but since the content
+it's pulling will have already been validated, the likelihood of a
+failure is drastically reduced.
+
+### Changes
+- The `vets-website` build script will no longer build the content
+  - **Possible exceptions:**
+	- We may need to update some things in the
+	  pre-built content such as references to the new cache-busted
+	  filenames for JS and CSS assets
+	- It should still be responsible for building the React pages,
+	  which would need to have the surrounding markup come from the
+	  templates
+- There will be a new repo for the content build process
+  - This will include scripts and templates
+- There will be a new Jenkins pipeline for building the content
+  - And everything that goes with this
+- There will be a (new?) S3 bucket which holds all the built content
+  - We currently do this with the cache, so this isn't much of a
+    deviation
+- The preview server will live in the new content-building repo
+- Sitemap accessibility tests will be done during the content build,
+  not the `vets-website` build
+- The `vets-website` Jenkins build will have the following changes
+  - There will be no cache fallback
+  - There will be no content caching
+  - There will be no sitemap accessibility testing
+  
+### Serving React apps
+`vets-website` should be responsible for making the pages where the
+React apps live, but not responsible for building the header, footer,
+and other surrounding markup. To make this possible, **I propose we
+have the content build produce an empty React page which the
```

Comment:
I don't really understand this section.  what is a "React page"?

"an empty React  page" --> one such page that gets reused for multiple React apps? or one for each React app?

Reply:
I added a glossary term for "React app page" and (hopefully :crossed_fingers:) clarified this a bit by using the terms "template" and "landing page."

The idea here is that the content build makes a template which the `vets-website` build can use to make landing pages for all its applications. This ensures:
- The content build doesn't have to know about the React apps
- The `vets-website` build doesn't have to know about the header, footer, or any other page chrome; its only concern is updating the references to the individual JS bundles and CSS files to bootstrap the application
  - I still have to dig into how we do this right now, though, to ensure we don't have a cleverer solution already